### DISCUSSION TITLE
(Interactive ) Error Boundary like on next.js?

### DISCUSSION THREAD

**Discussion by Jarrodsz** [remix-run/remix#4588] [Proposals]
As a rails developer background Im really missing an error console in Remix.
It would be great if remix would catch build errors. 

![166362901-5c2059fe-91a5-4290-b476-3cafbdd7d455](https://user-images.githubusercontent.com/90145421/201520493-4a553a9c-3892-4bcc-ad9b-6ef51b164537.png)

In rails we have error_console with interactive code debugging in the browser.
![80a24ccab45d803d](https://user-images.githubusercontent.com/90145421/201520548-623b88a8-3595-47ff-82ea-67744b5f6170.png)

The way this is really handy one can directly test code, variables etc in the browser upon errors.

**Comment by machour** [comment]
Hard +1 here.
While introducing people to remix, I keep getting this message: "when I change something in code it doesn't appear on the browser 😵‍💫"

Solution is always to check the server logs, or restart the server to see the compilation error.
Having an overlay would greatly improve DX

**Reply by kiliman** [reply]
I created a patch to do this. Nothing fancy, but it does show the error message. I should create an actual PR for this.

Here's a video showing what it looks like: 

https://www.loom.com/share/39c6bd8ad3a94e2bbb9888b158f35c27

**Reply by kiliman** [reply]
<img width="1010" alt="image" src="https://user-images.githubusercontent.com/47168/210260731-ec6c990d-e18f-44f9-aaf2-44cc40416297.png">

**Reply by machour** [reply]
oh nice, do you have a gist somewhere?

**Reply by Jarrodsz** [reply]
Is this patch available somewhere? find myself making lots of errors as newbie and sometimes have to check the terminal to see if things broke when not obvious. This would provide a single source of truth to check if something is broken the browser.

**Reply by kiliman** [reply]
Finally got around to creating a repo.

https://github.com/kiliman/remix-build-error

**Reply by Jarrodsz** [reply]
> Finally got around to creating a repo.
> 
> https://github.com/kiliman/remix-build-error

Awesome this helps a lot for newbies like me making code mistakes just to often ;)

**Comment by Jarrodsz** [comment]
Yes please make this happen!

**Comment by brophdawg11** [comment]
This discussion is being closed as part of the Remix v3 repository cleanup.

Remix v1/v2 was [upstreamed into React Router v7](https://remix.run/blog/wake-up-remix), so if this proposal is still relevant, please reopen or continue the discussion in the [React Router](https://github.com/remix-run/react-router) repository.
